### PR TITLE
Update `GesturesExtensions`

### PR DESCRIPTION
**Pull request by brminnick**
### Description of Change ###

This PR adds additional functionality to `GesturesExtensions`:
- Update XML Comments
- Add `numberOfClicksRequired` parameter to `BindClickGesture()` and `ClickGesture()`
- Add `direction` and `threshold` parameters to `BindSwipeGesture()` and `SwipeGesture()`
- Add `numberOfTapsRequired` parameter to `BindTapGesture()` and `TapGesture()`
- Add `touchPoints` parameter to `PanGesture()`

This PR also allows users to action on the gesture, e.g. `Action onClicked`, `Action<SwipedEventArgs>? onSwiped`. This a current bug where the `Action` passed in as a parameter was invoked immediately and not assigned to the GestureRecognizer.

 ### PR Checklist ###
 - [x] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [x] Has tests (if omitted, state reason in description)
 - [x] Has samples (if omitted, state reason in description)
 - [x] Rebased on top of `main` at time of PR
 - [x] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui.Markup/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [x] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pulls/XYZ 

 ### Additional information ###

 

This PR also updates the overrides in `BindableLayoutExtensions.EmptyViewTemplate` and `BindableLayoutExtensions.ItemTemplate` to leverage existing methods to avoid duplicating code.

### CODE REVIEW

File: src/CommunityToolkit.Maui.Markup/GesturesExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+	/// <returns><paramref name="gestureElement"/></returns>
+	public static TGestureElement PanGesture<TGestureElement>(this TGestureElement gestureElement,
+																Action<PanUpdatedEventArgs>? onPanUpdated = null,
+																int? touchPoints = null) where TGestureElement : IGestureRecognizers
+	{
+		var gestureRecognizer = new PanGestureRecognizer();
+
+		if (onPanUpdated is not null)
+		{
+			gestureRecognizer.PanUpdated += (sender, e) => onPanUpdated?.Invoke(e); ;
```
Comment (bijington) [inline src/CommunityToolkit.Maui.Markup/GesturesExtensions.cs:136]:
```suggestion
			gestureRecognizer.PanUpdated += (sender, e) => onPanUpdated?.Invoke(e);
```

File: src/CommunityToolkit.Maui.Markup.UnitTests/GesturesExtensionsTests.cs [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+	[Test]
+	public void ClickGesture()
+	{
+		const int numberOfClicks = 2;
+		int clicks = 0;
+
+		var gestureElement = new TGestureElement();
+
+		gestureElement.ClickGesture(() => clicks++, 2);
```
Comment (bijington) [inline src/CommunityToolkit.Maui.Markup.UnitTests/GesturesExtensionsTests.cs:82]:
```suggestion
		gestureElement.ClickGesture(() => clicks++, numberOfClicks);
```

**Review by bijington — COMMENTED** [review]
Thanks @brminnick it looks good! There is just the topic we discussed offline of checking whether the event subscription introduces any memory issues

**Comment by brminnick** [comment]
Thanks Sean! I found that we were inducing a memory leak by passing an `Action` into the event handler inside of a static class. 

The best solution I could find is to instead pass in an `EventHandler`, eg `EventHandler<PanUpdatedEventArgs>`. This ensures that the EventHandler method lives in the same class as the control and that it will be disposed accordingly when the control is disposed.

👇 Here's two ways users can now consume it

```cs
Content = new Label().PanGesture(OnPan);

async void OnPan(object? sender, PanUpdatedEventArgs e)
{
    await DisplayAlert("Pan Gesture Triggered", "", "Ok");
}
```

```cs
Content = new Label().PanGesture(
  async (sender, e) => await DisplayAlert("Pan Gesture Triggered", "", "Ok"));
```

Let me know what you think! If you approve, we can merge this PR and I write the docs before next week 🙌

**Review by bijington — APPROVED** [review]

**Comment by bijington** [comment]
> Thanks Shaun! I found that we were inducing a memory leak by passing an `Action` into the event handler inside of a static class.
> 
> The best solution I could find is to instead pass in an `EventHandler`, eg `EventHandler<PanUpdatedEventArgs>`. This ensures that the EventHandler method lives in the same class as the control and that it will be disposed accordingly when the control is disposed.
> 
> 👇 Here's two ways users can now consume it
> 
> ```cs
> Content = new Label().PanGesture(OnPan);
> 
> async void OnPan(object? sender, PanUpdatedEventArgs e)
> {
>     await DisplayAlert("Pan Gesture Triggered", "", "Ok");
> }
> ```
> 
>  ```cs
> Content = new Label().PanGesture(
>   async (sender, e) => await DisplayAlert("Pan Gesture Triggered", "", "Ok"));
> ```
> 
> Let me know what you think! If you approve, we can merge this PR and I write the docs before next week 🙌

That approach looks fine to me.

**Pull request merged**